### PR TITLE
Show service credentials in detail view

### LINKED ISSUE #2328: Allow the user to get service credentials
**Is your feature request related to a problem? Please describe.**
Since we are developing: https://github.com/epinio/epinio/issues/2037 that is: exposing services created with epinio to the user; we are required also to equip the user with the credentials needed to access the exposed services.
Currently, there is no such command that the user can invoke to get the credentials for an instance of a service.
The only way the user can obtain the credentials is to search into some kubernetes secretes; since epinio is designed to provide a layer to hide kubernetes concepts, the current available method seems to be logically unacceptable. 

**Describe the solution you'd like**
I would like, as user, a command such as:

`epinio service show-credentials mysql`

### PR DESCRIPTION
**Pull request by andreas-kupries**
Fix #2328

### CODE REVIEW

File: acceptance/services_test.go [hunk 1]
```diff
@@ -150,7 +150,35 @@ var _ = Describe("Services", LService, func() {
...
+		It("shows a service which has credentials", func() {
+			By("show it")
+			out, err := env.Epinio("", "service", "show", service)
+			Expect(err).ToNot(HaveOccurred(), out)
+			Expect(out).To(ContainSubstring("Showing Service"))
+			Expect(out).To(ContainSubstring("No settings"))
+			Expect(out).To(ContainSubstring("Credentials:"))
+			Expect(out).To(
+				HaveATable(
+					WithHeaders("KEY", "VALUE"),
```
Comment (giubacc) [inline acceptance/services_test.go:173]: I expect here to see also the username, not only the password; shouldn't the username key/value checked here as well?

File: internal/cli/usercmd/configuration.go [hunk 2]
```diff
@@ -500,7 +500,7 @@ func transformForDisplay(v string) string {
 	// Consider: Count and truncate by runes, not bytes.
 	// - https://pkg.go.dev/unicode/utf8@go1.20.5#RuneCountInString
 	// - https://go.dev/blog/strings (Libraries, foreach/range)
-	limit := 30
+	limit := 70
```
Comment (giubacc) [inline internal/cli/usercmd/configuration.go:503]: what is the meaning of this limit? oveflowing this value is something tested (or that makes sense to test) ?

File: internal/api/v1/service/show.go [hunk 3]
```diff
@@ -49,9 +52,32 @@ func Show(c *gin.Context) apierror.APIErrors {
 		return apierror.InternalError(err)
 	}
 
+	serviceConfigurations, err := configurations.ForService(ctx, cluster, service)
+	if err != nil {
+		return apierror.InternalError(err)
+	}
+
+	if len(serviceConfigurations) > 0 {
```
Comment (giubacc) [inline internal/api/v1/service/show.go:60]: hard to grok these bits; can you please explain what are we doing here?

File: acceptance/services_test.go (hunk 1 above)
Comment (andreas-kupries) [reply acceptance/services_test.go:173]: The change shows all the keys and their values in all the associated K secrets of the service.
The keys in the mysql service secrets do not provide any kind of user information.
I suspect that the helm chart (a bitnami one) uses a fixed username.

File: internal/cli/usercmd/configuration.go (hunk 2 above)
Comment (andreas-kupries) [reply internal/cli/usercmd/configuration.go:503]: Values larger than that are shown truncated (*). See https://github.com/epinio/epinio/pull/2513/files#diff-db92da3c647e894be7caec46cf94133def888e5c77dbc5a1109a4b9ec6b1614f for an affected test (configurations).

(*) They are likely multi-line as well, something the table formatting will have trouble with.

File: internal/api/v1/service/show.go (hunk 3 above)
Comment (andreas-kupries) [reply internal/api/v1/service/show.go:60]: I guess you are looking at the inner for loop with the `j` increments ?
Will add code comments.

Comment (andreas-kupries) [reply internal/api/v1/service/show.go:68]: Done with commit b82a5701e8caed448d3e43df07cbe2e35943e01c

**Comment by giubacc** [comment]
Generally lgtm; The codecov is not so good, there are paths not covered in the patch, probably erroneous scenarios. 
If this is not an issue, happy to approve.

**Comment by andreas-kupries** [comment]
I will check codecov and see if I can cover more by tests. Not sure yet. In general it might be error paths which we do/can not trigger.

**Comment by andreas-kupries** [comment]
> I will check codecov and see if I can cover more by tests. Not sure yet. In general it might be error paths which we do/can not trigger.

@giubacc 

Having looked things over the majority of coverage issues are indeed error paths we are simply not triggering in our tests.
And triggering looks to be difficult, likely requiring mocks and such. The one non-error case is the handling of conflicting keys. For that we need a service chart which actually has such, and I do not know of any. (The code was written because I have no proof there is no such chart).

So, no tests for any of these, and while not nice, IMHO good enough.

Please approve.

**Review by giubacc — APPROVED** [review]

**Comment by thehejik** [comment]
Verified in `v1.10.0-rc1`, in `v1.9.0` the credentials were not displayed.

```
$ epinio-1.10 service show mysql-sample

🚢  Showing Service...

✔️  Details:
|       KEY       |                                     VALUE                                      |
|-----------------|--------------------------------------------------------------------------------|
| Name            | mysql-sample                                                                   |
| Created         | 2023-09-08 10:16:30 +0200 CEST                                                 |
| Catalog Service | mysql-dev                                                                      |
| Version         | 8.0.31                                                                         |
| Status          | deployed                                                                       |
| Used-By         | sample-app                                                                     |
| Internal Routes | xc11fadea2478b7bb684678d6d8a8-mysql-headless.workspace.svc.cluster.local:3306, |
|                 | xc11fadea2478b7bb684678d6d8a8-mysql.workspace.svc.cluster.local:3306           |

⚠️  No settings

✔️  Credentials:
|         KEY         |   VALUE    |
|---------------------|------------|
| mysql-password      | JblvTfOSnd |
| mysql-root-password | MObIy5IsCD |
```

**Comment by mmartin24** [comment]
rechecked in `1.10.0` and ok:

```

🚢  Showing Service...

✔️  Details:
|       KEY       |                                     VALUE                                      |
|-----------------|--------------------------------------------------------------------------------|
| Name            | svc-mysql                                                                      |
| Created         | 2023-09-25 13:24:59 +0200 CEST                                                 |
| Catalog Service | mysql-dev                                                                      |
| Version         | 8.0.31                                                                         |
| Status          | deployed                                                                       |
| Used-By         | sampleapp2                                                                     |
| Internal Routes | xb097c63b2fb47a525229d1fc3003-mysql-headless.workspace.svc.cluster.local:3306, |
|                 | xb097c63b2fb47a525229d1fc3003-mysql.workspace.svc.cluster.local:3306           |

⚠️  No settings

✔️  Credentials:
|         KEY         |   VALUE    |
|---------------------|------------|
| mysql-password      | pYuMEYQlab |
| mysql-root-password | BYGtnfHhlA |
➜  Downloads epinio info                  

✔️  Epinio Environment
Platform: k3s
Kubernetes Version: v1.25.4+k3s1
Epinio Server Version: v1.10.0
Epinio Client Version: v1.10.0

```

**Pull request merged**